### PR TITLE
[DDO-2904] Expose metrics from Sherlock's CiRuns

### PR DESCRIPTION
**Pull request by jack-r-warren**
It exposes a count of completions and a sum of seconds spent. Both can be broken down by repo, workflow file, attempt number, and outcome.

These are ephemeral Sherlock metrics, more similar to the PagerDuty request metrics (that I think I never demo-ed, but they're there) than the accelerate metrics. Rather than being calculated on startup and reported as "LastValue" they're sent as requests come in and reported as "Count" or "Sum" (thus being aggregated with existing values Thanos has stored, not replacing them).

## Testing

I ran Sherlock and the webhook proxy locally. Then I ran Sherlock's [test-auth-transparency.yaml](https://github.com/broadinstitute/sherlock/blob/main/.github/workflows/test-auth-transparency.yaml) workflow three times. The middle invocation was me retrying the first.

Here's what the webhook proxy printed out:

```
2023/06/09 18:30:57 sherlockClient.CiRuns.PutAPIV2CiRunsSelector(): created CiRun 821, 'queued'
2023/06/09 18:31:06 sherlockClient.CiRuns.PutAPIV2CiRunsSelector(): edited CiRun 821, 'in_progress'
2023/06/09 18:31:10 sherlockClient.CiRuns.PutAPIV2CiRunsSelector(): edited CiRun 821, 'success'
2023/06/09 18:31:43 sherlockClient.CiRuns.PutAPIV2CiRunsSelector(): created CiRun 822, 'in_progress'
2023/06/09 18:31:50 sherlockClient.CiRuns.PutAPIV2CiRunsSelector(): edited CiRun 822, 'success'
2023/06/09 18:32:27 sherlockClient.CiRuns.PutAPIV2CiRunsSelector(): created CiRun 823, 'queued'
2023/06/09 18:32:36 sherlockClient.CiRuns.PutAPIV2CiRunsSelector(): edited CiRun 823, 'in_progress'
2023/06/09 18:32:42 sherlockClient.CiRuns.PutAPIV2CiRunsSelector(): edited CiRun 823, 'success'
```

(Note that retry doesn't appear to get queued, it goes straight to in_progress. No matter, just interesting.)

Sherlock printed this out (ignore the timestamps, they're in UTC not local time):

```
sherlock_1  | 10:30PM DBG enforcing *v2models.CiRun selector uniqueness across 1 selectors ([github-actions/broadinstitute/sherlock/5226565377/1]); handleConflicts provided=false
sherlock_1  | 10:30PM INF GIN  | 201 |      14.9895ms |      172.20.0.1 |        fake@broadinstitute.org | PUT     /api/v2/ci-runs/github-actions/broadinstitute/sherlock/5226565377/1
sherlock_1  | 10:31PM INF GIN  | 200 |    16.005375ms |      172.20.0.1 |        fake@broadinstitute.org | PUT     /api/v2/ci-runs/github-actions/broadinstitute/sherlock/5226565377/1
sherlock_1  | 10:31PM INF GIN  | 200 |    10.130083ms |      172.20.0.1 |        fake@broadinstitute.org | PUT     /api/v2/ci-runs/github-actions/broadinstitute/sherlock/5226565377/1
sherlock_1  | 10:31PM DBG enforcing *v2models.CiRun selector uniqueness across 1 selectors ([github-actions/broadinstitute/sherlock/5226565377/2]); handleConflicts provided=false
sherlock_1  | 10:31PM INF GIN  | 201 |    10.542625ms |      172.20.0.1 |        fake@broadinstitute.org | PUT     /api/v2/ci-runs/github-actions/broadinstitute/sherlock/5226565377/2
sherlock_1  | 10:31PM INF GIN  | 200 |        13.63ms |      172.20.0.1 |        fake@broadinstitute.org | PUT     /api/v2/ci-runs/github-actions/broadinstitute/sherlock/5226565377/2
sherlock_1  | 10:32PM DBG enforcing *v2models.CiRun selector uniqueness across 1 selectors ([github-actions/broadinstitute/sherlock/5226574705/1]); handleConflicts provided=false
sherlock_1  | 10:32PM INF GIN  | 201 |    12.403583ms |      172.20.0.1 |        fake@broadinstitute.org | PUT     /api/v2/ci-runs/github-actions/broadinstitute/sherlock/5226574705/1
sherlock_1  | 10:32PM INF GIN  | 200 |     12.13475ms |      172.20.0.1 |        fake@broadinstitute.org | PUT     /api/v2/ci-runs/github-actions/broadinstitute/sherlock/5226574705/1
sherlock_1  | 10:32PM INF GIN  | 200 |    10.441833ms |      172.20.0.1 |        fake@broadinstitute.org | PUT     /api/v2/ci-runs/github-actions/broadinstitute/sherlock/5226574705/1

```

(Note that the 201 responses correspond to the creations, while 200 correspond to the edits)

Then, visiting Sherlock's metrics, we see this at the bottom:

```
# HELP sherlock_v2_github_actions_completion_count count of completed GitHub Actions reported to Sherlock
# TYPE sherlock_v2_github_actions_completion_count counter
sherlock_v2_github_actions_completion_count{attempt_number="1",github_repo="broadinstitute/sherlock",outcome="success",workflow_file=".github/workflows/test-auth-transparency.yaml"} 2
sherlock_v2_github_actions_completion_count{attempt_number="2",github_repo="broadinstitute/sherlock",outcome="success",workflow_file=".github/workflows/test-auth-transparency.yaml"} 1
# HELP sherlock_v2_github_actions_duration_sum count of seconds spent by GitHub Actions reported to Sherlock
# TYPE sherlock_v2_github_actions_duration_sum counter
sherlock_v2_github_actions_duration_sum{attempt_number="1",github_repo="broadinstitute/sherlock",outcome="success",workflow_file=".github/workflows/test-auth-transparency.yaml"} 28
sherlock_v2_github_actions_duration_sum{attempt_number="2",github_repo="broadinstitute/sherlock",outcome="success",workflow_file=".github/workflows/test-auth-transparency.yaml"} 17
```

In other words, exactly what I need to make a GHA dashboard.

## Risk

I'm modifying Sherlock's generic create and edit methods on top of Gorm, but I'm doing so in a way that A) can't error and B) is extremely lightweight and C) only impacts CiRun. This looks super low risk to me.

### CODE REVIEW

**Review by jyang-broad — APPROVED** [review]
:stamp: i think it makes sense.
